### ISSUE TITLE
Clojure-lsp causing a UI freeze, OOM errors, or just using a large amount of memory (2+ GB)

### ISSUE THREAD

**Issue by bpringe** [BetterThanTomorrow/calva#990]
Things like this have been reported by one or two users. If you experience this, please let us know the details here and add the specs of your system.

* OS
* RAM
* VS Code version
* Calva version

**Comment by Nillkki** [comment]
Hi!
I'm having this issue! UI keeps freezing. I believe this is due to clojure-lsp using 2GB of memory. My memory usage is generally high due to other programs but 2GB seems high. I'm only learning Clojure. Code I'm experiencing this with can be found: https://github.com/iloveponies/structured-data

The issue goes away for awhile if I restart VS Code!

* OS: macOS 11.1(Darwin x64 20.2.0)
* RAM: 16GB
* VS Code version: 1.52.1
* Calva version: 2.0.157

**Comment by bpringe** [comment]
Thanks so much for reporting and providing a repo! Does this happen with other projects too or just that one? If it does happen with others, is it consistent per project?

**Comment by bpringe** [comment]
I want to link this related issue: https://github.com/clojure-lsp/clojure-lsp/issues/229. Though, the UI freeze thing may be more of a CPU issue than a memory issue.

@Nillkki Have you checked your system resources when the UI freezes? Is it just the VS Code UI that freezes or your whole OS UI? If it's not the whole OS UI, would you mind checking your CPU and memory usage next time a freeze happens and reporting that here?

**Comment by bpringe** [comment]
I opened the linked project and monitored my system resources. CPU usage goes pretty high for a bit but no freezing occurs. I also compared the CPU usage with another project at startup (also having removed the clj-kondo and lsp files). The linked project usage is at the top and the other is at the bottom.

![image](https://user-images.githubusercontent.com/12722744/106796373-1a331f80-6610-11eb-8e68-afdab5f68029.png)

![image](https://user-images.githubusercontent.com/12722744/106796429-2cad5900-6610-11eb-98a6-1d3c67b68428.png)
